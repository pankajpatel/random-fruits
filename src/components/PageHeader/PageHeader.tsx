import { ChangeEvent, ReactNode, useContext } from 'react';
import { Header } from '@app/ds/Header';
import { Select } from '@app/ds/Select';
import { LocaleContext } from '@app/contexts/Locale';
import styled from 'styled-components';

const H2 = styled.h2`
  margin-top: 1rem;
  margin-bottom: 1rem;
  display: flex;
  align-items: center;
  & > * {
    margin-right: 0.5rem;
  }
`;

export const PageHeader = ({
  children,
}: {
  children: ReactNode;
}): JSX.Element => {
  const { locale, setLocale } = useContext(LocaleContext);
  return (
    <Header>
      <H2>{children}</H2>
      <Select
        value={locale}
        onChange={(e: ChangeEvent<HTMLSelectElement>) => {
          setLocale(e.target.value as Locale);
        }}
      >
        <option value="en">EN ğŸ‡¬ğŸ‡§</option>
        <option value="fr">FR ğŸ‡«ğŸ‡·</option>
      </Select>
    </Header>
  );
};
