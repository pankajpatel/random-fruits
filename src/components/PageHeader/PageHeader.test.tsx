import { render } from "@testing-library/react";
import { PageHeader } from "./PageHeader";

describe("PageHeader", () => {
  it("should render without crashing", () => {
    const { container } = render(<PageHeader>Test</PageHeader>);
    expect(container).toMatchSnapshot();
  })
});
