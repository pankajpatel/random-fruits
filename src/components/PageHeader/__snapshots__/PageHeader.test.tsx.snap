// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageHeader should render without crashing 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  border-bottom: 1px solid rgba(0,0,0,0.2);
  padding: 0 0.5rem;
  margin-bottom: 1.5rem;
}

.c2 {
  font-size: 1.2rem;
  padding: 0.5rem;
  border: 1px solid #ccc;
  border-radius: 0.25rem;
  width: 100%;
}

.c1 {
  margin-top: 1rem;
  margin-bottom: 1rem;
}

<div>
  <header
    class="c0"
  >
    <h2
      class="c1"
    >
      Test
    </h2>
    <div
      class=""
    >
      <select
        class="c2"
      >
        <option
          value="en"
        >
          ðŸ‡¬ðŸ‡§
        </option>
        <option
          value="fr"
        >
          ðŸ‡«ðŸ‡·
        </option>
      </select>
    </div>
  </header>
</div>
`;
